server/controllers/user.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const userModel = require('../model/user');
+const Logger = require('../services/logger_services');
+
+const app = express();
+
+const logger = new Logger('app');
+
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+const newUserValidation = (user) => {
+  const userNameValidationRegEx = /^([a-z]|[0-9]|-|_)+$/;
+  const isUserNameValid = userNameValidationRegEx.test(user.userName);
+  if (!isUserNameValid) {
+    throw new Error('username is invalid');
+  }
+  if (!user.password) {
+    throw new Error('password field is empty');
+  }
+  if ((user.password).length < 8) {
+    throw new Error('password field should have 8 or more digits and characters');
+  }
+  if (!user.firstName) {
+    throw new Error('firstname field is empty');
+  }
+  if (!user.lastName) {
+    throw new Error('lastName field is empty');
+  }
+  if (!user.address) {
+    throw new Error('address field is empty');
+  }
+  if (!!(typeof (user.age) === 'number') || !(user.age >= 0)) {
```

Comment:
!!(typeof (user.age) === 'number') means ==> (typeof (user.age) === 'number'), so the check is broken?
and simplify the if